challenges/templates/challenges/index.html

```diff
@@ -2,47 +2,78 @@
 
 {% load static %}
 {% load utils %}
-
 {% block extra-js %}
     <script src="{% static 'js/index.js' %}"></script>
     <script src="{% static 'js/plotly-latest.min.js' %}"></script>
     <script>
-    document.addEventListener('DOMContentLoaded', function(){
-    var data = [{
-        x: [
-            {% for d in days %}
-            '{{ d|date:"Y-m-d" }}',
-            {% endfor %}
-        ],
-        y: [
-            {% for d in days %}
-            {{ submissions_per_day|get_item:d }},
-            {% endfor %}
-        ],
-        type: 'bar'
-    }];
-    var layout = {
-        xaxis: {
-            title: {
-                text: 'Data',
-            }
-        },
-        yaxis: {
-            title: {
-                text: '# Submissões',
-            }
-        },
-    }
-
-    Plotly.newPlot('plotDias', data, layout);
-});
-    </script>
+     document.addEventListener('DOMContentLoaded', function(){
+     var data = [{
+         x: [
+             {% for d in days %}
+             '{{ d|date:"Y-m-d" }}',
+             {% endfor %}
+         ],
+         y: [
+             {% for d in days %}
+             {{ submissions_per_day|get_item:d }},
+             {% endfor %}
+         ],
+         type: 'bar'
+     }];
+     var layout = {
+         xaxis: {
+             title: {
+                 text: 'Data',
+             }
+         },
+         yaxis: {
+             title: {
+                 text: '# Submissões',
+             }
+         },
+     }
+     Plotly.newPlot('plotDias', data, layout);
+     });
+</script>
 {% endblock %}
 
 {% block pagecontent %}
 
 <h1>Acompanhamento</h1>
 
+<h2>Progresso</h2>
+
+<p>
+A seguir você pode acompanhar o seu progresso nos desafios por tags.
+</p>
+
+<div >
+    <div>    
+        <link rel="stylesheet" type="text/css" href="./static/css/loading-bar.css"/>
+        <script type="text/javascript" src="/static/js/loading-bar.js"></script>
+        <div style="width: 100%; display: table;"><div style="display: table-row"></div>
+        {%for t in prog_tags%}
+        {%with tag=prog_tags|get_item:t%}
+        {%with pct=tag|get_item:'pct'%}
```

Comment:
Please use less with tags: https://docs.djangoproject.com/en/2.2/ref/templates/builtins/#with
You can define multiple variables in the same 'with' templatetag:
{% with a=1 b=2 c=3 %}

Also, please add a space after %.

Reply:
Done!